### PR TITLE
feat: add show_legend_state

### PR DESCRIPTION
**Pull request by ljmerza**
https://github.com/kalkih/mini-graph-card/issues/1081

image

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -338,17 +338,29 @@ class MiniGraphCard extends LitElement {
...
-          <div class="graph__legend__item"
-            @click=${e => this.handlePopup(e, this.entity[entity.index])}
-            @mouseenter=${() => this.setTooltip(entity.index, -1, this.getEntityState(entity.index), 'Current')}
-            @mouseleave=${() => (this.tooltip = {})}>
-            ${this.renderIndicator(this.entity[entity.index].state, entity.index)}
-            <span class="ellipsis">${this.computeName(entity.index)}</span>
-          </div>
-        `)}
+        ${this.visibleLegends.map((entity) => {
+    let legend = this.computeName(entity.index);
```
Comment (akloeckner) [inline src/main.js:346]: Please properly indent the code.

File: src/main.js [hunk 2]
```diff
@@ -338,17 +338,29 @@ class MiniGraphCard extends LitElement {
...
+    const state = this.getEntityState(entity.index);
+
+    const { show_legend_state } = this.config.entities[entity.index];
+    if (show_legend_state) {
+      legend += ` (${this.computeState(state)}${this.computeUom(entity.index)})`;
```
Comment (akloeckner) [inline src/main.js:351]: Can you put the state in an additional `span` with a suitable class? (If there already are suitable classes for state and UOM, you could maybe reuse them.)

I think there is a CSS property then to add the braces to the content. 

That way users could use `card-mod` to style the appearance of the state as they wish.

Please make sure the defaults look good in different situations, e.g. long and short state names.

**Review by akloeckner — CHANGES_REQUESTED** [review]
Thanks for the suggestion! Looks like a useful feature.

I added a few remarks. I'll still have to test the feature before accepting it.

Please also direct your PR to the `dev` branch.

File: src/main.js (hunk 2 above)
Comment (ildar170975) [reply src/main.js:351]: > put the state in an additional span

Absolutely correct.

Also, please think about an order: in some cases a user may want to have UoM before a value.

Comment (akloeckner) [reply src/main.js:351]: I think, a separate `span` should do the trick for that as well, together with the CSS `direction` property.

**Comment by JulienDeveaux** [comment]
@akloeckner don't forget to close this one 😉

**Comment by ildar170975** [comment]
Closed in favour of https://github.com/kalkih/mini-graph-card/pull/1173

**Pull request closed without merging**